HeimdallrTests/Core/HeimdallrSpec.swift

```diff
@@ -670,6 +670,36 @@ class HeimdallrSpec: QuickSpec {
                     beforeEach {
                         OHHTTPStubs.stubRequestsPassingTest({ request in
                             return (request.URL!.absoluteString == "http://rheinfabrik.de")
+                            }, withStubResponse: { request in
+                                return OHHTTPStubsResponse(data: NSData(contentsOfFile: self.bundle.pathForResource("request-valid-norefresh", ofType: "json")!)!, statusCode: 200, headers: [ "Content-Type": "application/json" ])
+                        })
+                        
+                        waitUntil { done in
+                            heimdallr.authenticateRequest(request) { result = $0; done() }
+                        }
+                    }
+                    
+                    it("succeeds") {
+                        expect(result?.value).toNot(beNil())
+                    }
+                    
+                    it("attempts to parse the fresh token") {
+                        expect(accessTokenParser.timesCalled).to(equal(2))
+                    }
+                    
+                    it("authenticates the request using the resource request authenticator") {
+                        expect(result?.value?.valueForHTTPHeaderField("MockAuthorized")).to(equal("totally"))
+                    }
+                    
+                    it("keeps the original refresh token") {
+                        expect(accessTokenStore.retrieveAccessToken()?.refreshToken).to(equal("ZmVhMThjYzUyZDM3MmIzNDcyMDMyMzc2MzhmYTg4YWM0MWYyYmQxZmFlMTE2Mzk0MWY5YTk1YWQ4ZDBmYzIxZA"))
+                    }
+                }
+                
+                context("when refreshing the access token succeeds and overrides the refresh token") {
```

Comment:
I think the contexts can be rearranged so that this one and the one above read a bit nicer. I would propose something like this: 

```swift
context("when refreshing the access token succeeds") {
  context("and the response does not contain a refresh token") {

  }
  
  context("and the response does contain a refresh token") {
  
  }
}
```